### PR TITLE
raise worker errors in launcher

### PR DESCRIPTION
**Pull request by pmcurtin**
[draft]

### CODE REVIEW

**Comment by pmcurtin** [comment]
@apoorvkh ok, those changes look nice!

File: src/torchrunx/agent.py [hunk 1]
```diff
@@ -93,13 +93,14 @@ def entrypoint(serialized_worker_args: bytes):
...
-    sys.stdout.flush()
-    sys.stderr.flush()
-
-    return r
+    try:
+        return worker_args.function()
+    except Exception as e:
+        logger.error(e)
+        return WorkerException(exception=e)
+    finally:
```
Comment (apoorvkh) [inline src/torchrunx/agent.py:101]: except: … raise
finally: return return_object

Comment (pmcurtin) [reply src/torchrunx/agent.py:101]: Is there any point to raising the exception?

Comment (apoorvkh) [reply src/torchrunx/agent.py:101]: I guess just that actually raising the exception (which should be logged via redirect_stderr anyway) is more "proper" than logger.log(exception)? There may be a difference in the log output?

**Review by apoorvkh — APPROVED** [review]

**Pull request merged**